src/distrs/pois.jl

```diff
@@ -19,6 +19,13 @@ poislogpdf(λ::T, x::T) where {T <: Real} = xlogy(x, λ) - λ - loggamma(x + 1)
 
 poislogpdf(λ::Number, x::Number) = poislogpdf(promote(float(λ), x)...)
```

Comment:
I think I get why this is done btw: `loggamma(::ForwardDiff.Dual, ::Int)` won't have a diff-rule defined for it.

Reply:
Actually, it doesn't seem like these have rules defined in DiffRules.jl!

Reply:
There is no DiffRule defined for `loggamma` and `gamma_inc` anyway so it shouldn't matter at the moment :shrug: I added ChainRules definitions to SpecialFunctions (https://github.com/JuliaMath/SpecialFunctions.jl/pull/305) but currently they can't be used with Zygote due to https://github.com/FluxML/Zygote.jl/issues/873.

So can we avoid these conversions if possible for now and only add them later if needed?

Reply:
I'll have to look at it later, but it seems like there are particular execution branches (depending on the values) which ForwardDiff.jl fails to automatically obtain derivatives for. So when I just tried a couple of values everything seemed to work, but when I let HMC loose on a distribution using this function under the hood, it eventually runs into a `MethodError`.

Reply:
Haha, man, you're always like one step ahead: I just looked at your PR and that's exactly the "solution" i arrived at literally 10mins ago, i.e. we can only take derivative wrt. second argument but that's also what we care about so w/e.

> There is no DiffRule defined for loggamma and gamma_inc anyway so it shouldn't matter at the moment 

But wouldn't adding those fix this issue? Same as you've done in ChainRules now?

Reply:
Sure, if I used the correct derivatives it should be fine :stuck_out_tongue: AFAICT the convention in DiffRules would be to set the derivatives of the first argument to NaN, if it is not implemented properly.